daemon/api_snaps.go

```diff
@@ -260,6 +261,47 @@ func (inst *snapInstruction) holdLevel() snapstate.HoldLevel {
 	}
 }
 
+func cleanSnapshotOptions(snapshotOpts *map[string]*snap.SnapshotOptions) {
+	cleanSnapshotOpts := map[string]*snap.SnapshotOptions{}
+	for name, options := range *snapshotOpts {
+		if !options.Unset() {
+			cleanSnapshotOpts[name] = options
+		}
+	}
+
+	*snapshotOpts = cleanSnapshotOpts
```

Comment:
I agree that taking a pointer and reassigning a map instead of returning it is unexpected. But this could just delete the unwanted map entries instead of allocating a new map

```suggestion
func cleanSnapshotOptions(snapshotOpts map[string]*snap.SnapshotOptions) {
	for name, options := range snapshotOpts {
		if options.Unset() {
			delete(snapshotOpts, name)
		}
	}
```

Reply:
Thanks, implemented your proposal.